component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java

```diff
@@ -198,7 +230,15 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
                     if (log.isDebugEnabled()) {
                         log.debug("User has not enabled TOTP: " + username);
                     }
-                    Map<String, String> claims = TOTPKeyGenerator.generateClaims(username, false, context);
+                    Map<String, String> claims;
+                    if (isFederatedUser) {
+                        String storedSecretKey = totpSecretKeyDAO.getTOTPSecretKeyOfFederatedUser(userId);
+                        claims = TOTPKeyGenerator.generateClaimsForFedUser(username, tenantDomain, storedSecretKey,
+                                false, context);
```

Comment:
Is there any possibility to have a null value for storedSecretKey? If the storedSecretKey is null for the specified federated user-id what could be the behavior?

Reply:
Fixed https://github.com/wso2-extensions/identity-outbound-auth-totp/pull/94/commits/01e2f562f9e60129c3810e75f2cb8f16020632bc

Reply:
If it's null it will re-generate the secret key[1]
[1] https://github.com/wso2-extensions/identity-outbound-auth-totp/pull/94/files#diff-ae070d51d9ab9a1117f87cdfc7a7fb153bfbbd4afd9fe1549244763945cb2786R130-R132